src/javascript/app_2/pages/trading/components/elements/sidebar.jsx

```diff
@@ -0,0 +1,138 @@
+                </div>
+              ))}
+          </div>
+        );
+    }
+}
+
+class SidebarSubmenu extends React.PureComponent {
+    constructor(props) {
+        super(props);
+    }
+
+    render() {
+        const getHref = item_id => `#${item_id}`;
+        const { isOpen } = this.props;
+        return (
+            <div className="sidebar-submenu">
+                <ul>
+                    {this.props.items.map((item, idx) => (
+                      <li key={idx} className={(item.submenu ? 'has-submenu' : '') + (item.id == this.props.selectedId ? 'selected' : '')}>
+                          { item.submenu &&
+                          <React.Fragment>
+                              <a id={item.id}
+                                 className={isOpen == item.id ? 'selected' : ''}
+                                 href={getHref(item.id)}
+                                 onClick={this.props.openSubmenu}>
+                                  {item.text}
+                                  </a>
+                              <ul className={isOpen == item.id ? 'collapsed' : 'compressed'}>
+                                  { item.submenu.map((submenu, idx_submenu) => (
+                                    <li key={idx_submenu}
+                                        className={submenu.id == this.props.selectedId ? 'active' : ''}>
+                                        <a id={submenu.id} onClick={this.props.onClick}
+                                           className={'submenu-item ' + (submenu.id == this.props.selectedId ? 'selected' : '')}
+                                           href={getHref(submenu.id)}>
+                                            {submenu.text}
+                                        </a>
+                                    </li>
+                                  ))}
+                              </ul>
+                          </React.Fragment>}
+                          { !item.submenu && item.text &&
+                          <a
+                            className={item.id == this.props.selectedId ? 'selected' : ''}
+                            id={item.id}
+                            onClick={this.props.onClick}
+                            href={getHref(item.id)}>{item.text}</a>}
+                      </li>
+                    ))}
+                </ul>
+            </div>
+        );
+    }
+}
+
+// Dummy values
+const dummyitems = [
+    {
+        id     : 'legal',
+        text   : 'Terms and Conditions',
+        submenu: [
+            { id: 'copyright', text: 'Copyright Section' },
+            { id: 'privacy',     text: 'Privacy Section' },
+        ],
+    },
+    { id: 'datafeed',  text: 'Data Feed Section' },
+];
+
+const contents = [
+  {
+    id: 'copyright',
+    cont: <div>copyright</div>
+  },
+    {
+        id: 'privacy',
+        cont: <div>privacy</div>
+    },
+    {
+        id: 'datafeed',
+        cont: <div>datafeed</div>,
+    }
+];
+
+Sidebar.defaultProps = {
```

Comment:
Comment out default props, otherwise it will end up in the final bundle.

Reply:
@aminroosta thanks, all done now